src/sentry/integrations/github/search.py

```diff
@@ -1,84 +1,67 @@
-from typing import Any
-
-from rest_framework.request import Request
 from rest_framework.response import Response
 
-from sentry.api.api_owners import ApiOwner
-from sentry.api.api_publish_status import ApiPublishStatus
 from sentry.api.base import control_silo_endpoint
-from sentry.integrations.api.bases.integration import IntegrationEndpoint
 from sentry.integrations.github.integration import GitHubIntegration, build_repository_query
 from sentry.integrations.github_enterprise.integration import GitHubEnterpriseIntegration
 from sentry.integrations.models.integration import Integration
-from sentry.organizations.services.organization import RpcOrganization
+from sentry.integrations.source_code_management.search import SourceCodeSearchEndpoint
 from sentry.shared_integrations.exceptions import ApiError
 
 
 @control_silo_endpoint
-class GithubSharedSearchEndpoint(IntegrationEndpoint):
-    owner = ApiOwner.ECOSYSTEM
-    publish_status = {
-        "GET": ApiPublishStatus.PRIVATE,
-    }
+class GithubSharedSearchEndpoint(SourceCodeSearchEndpoint):
     """NOTE: This endpoint is a shared search endpoint for Github and Github Enterprise integrations."""
 
-    def get(
-        self, request: Request, organization: RpcOrganization, integration_id: int, **kwds: Any
-    ) -> Response:
-        try:
-            integration = Integration.objects.get(
-                organizationintegration__organization_id=organization.id,
-                id=integration_id,
-            )
-        except Integration.DoesNotExist:
-            return Response(status=404)
+    @property
+    def repository_field(self):
+        return "repo"
+
+    @property
+    def integration_provider(self):
+        return None
 
-        field = request.GET.get("field")
-        query = request.GET.get("query")
-        if field is None:
-            return Response({"detail": "field is a required parameter"}, status=400)
-        if not query:
-            return Response({"detail": "query is a required parameter"}, status=400)
+    @property
+    def installation_class(self):
+        return (GitHubIntegration, GitHubEnterpriseIntegration)
```

Comment:
It would simplify things to remove the tuple return type from the parent class. We could do that there were a base class that `GitHubIntegration` and `GitHubEnterpriseIntegration` shared (and only them). Could you create a small one, just for this purpose? (Maybe "GenericGitHubIntegration", unless you want to rename "GitHubIntegration" and use that as the name for the base class.)